x-pack/plugins/threat_intelligence/public/modules/indicators/hooks/use_aggregated_indicators.ts

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import moment from 'moment';
+import { buildEsQuery, TimeRange } from '@kbn/es-query';
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { Subscription } from 'rxjs';
+import {
+  IEsSearchRequest,
+  IKibanaSearchResponse,
+  isCompleteResponse,
+  isErrorResponse,
+  TimeRangeBounds,
+} from '@kbn/data-plugin/common';
+import { RawIndicatorFieldId } from '../../../../common/types/indicator';
+import { useKibana } from '../../../hooks/use_kibana';
+import { DEFAULT_THREAT_INDEX_KEY } from '../../../../common/constants';
+import { DEFAULT_TIME_RANGE } from './use_filters/utils';
+
+export interface UseAggregatedIndicatorsParam {
+  timeRange?: TimeRange;
+}
+
+export interface UseAggregatedIndicatorsValue {
+  indicators: ChartSeries[];
+  onFieldChange: (field: string) => void;
+  dateRange: TimeRangeBounds;
+}
+
+export interface Aggregation {
+  doc_count: number;
+  key: string;
+  events: {
+    buckets: AggregationValue[];
+  };
+}
+
+export interface AggregationValue {
+  doc_count: number;
+  key: number;
+  key_as_string: string;
+}
+
+export interface ChartSeries {
+  x: string;
+  y: number;
+  g: string;
+}
+
+const THREAT_QUERY_BASE = 'event.type: indicator and event.category : threat';
```

Comment:
Looks like this could be put into `constants` or something similar as it is used in two different hooks now:)

Reply:
I had thought about this and I guess forgot to move it. Done!